src/mlpack/methods/neighbor_search/sort_policies/nearest_neighbor_sort.hpp

```diff
@@ -150,6 +150,21 @@ class NearestNeighborSort
       return DBL_MAX;
     return a + b;
   }
+
+  /**
+   * Return the given value relaxed.
+   *
+   * @param value Value to relax.
+   * @param epsilon Relative error (non-negative).
+   *
+   * @return double Value relaxed.
+   */
+  static inline double Relax(const double value, const double epsilon)
+  {
+    if (value == DBL_MAX)
+      return DBL_MAX;
+    return (1 / (1 + epsilon)) * value;
```

Comment:
Okay so as I understand it, for KNN we have a strong preference; go ahead with the current range. I agree with option 4 for KFN. In case of users who understand KFN but not approximation (which will be the case with most of the users who use command line tool and which is the case with couple of my friends), most of them will prefer percentage option as it is clear. I mean the option suggested by @rcurtin, if the user wants furthest neighbors that are at least x% of the distance as the true furthest neighbor, they just pass in "x/100". This way the concept of epsilon will be consistent from KNN to KFN and the users won't have to go through this when using the executable.